metric/noop/noop.go

```diff
@@ -0,0 +1,270 @@
+
+// Int64UpDownCounter is an OpenTelemetry UpDownCounter used to record int64
+// measurements. It produces no telemetry.
+type Int64UpDownCounter struct {
+	instrument.Int64UpDownCounter
+}
+
+// Add performs no operation.
+func (Int64UpDownCounter) Add(context.Context, int64, ...attribute.KeyValue) {}
+
+// Float64UpDownCounter is an OpenTelemetry UpDownCounter used to record
+// float64 measurements. It produces no telemetry.
+type Float64UpDownCounter struct {
+	instrument.Float64UpDownCounter
+}
+
+// Add performs no operation.
+func (Float64UpDownCounter) Add(context.Context, float64, ...attribute.KeyValue) {}
+
+// Int64Histogram is an OpenTelemetry Histogram used to record int64
+// measurements. It produces no telemetry.
+type Int64Histogram struct {
+	instrument.Int64Histogram
+}
+
+// Record performs no operation.
+func (Int64Histogram) Record(context.Context, int64, ...attribute.KeyValue) {}
+
+// Float64Histogram is an OpenTelemetry Histogram used to record float64
+// measurements. It produces no telemetry.
+type Float64Histogram struct {
+	instrument.Float64Histogram
+}
+
+// Record performs no operation.
+func (Float64Histogram) Record(context.Context, float64, ...attribute.KeyValue) {}
+
+// Int64ObservableCounter is an OpenTelemetry ObservableCounter used to record
+// int64 measurements. It produces no telemetry.
+type Int64ObservableCounter struct {
+	instrument.Int64ObservableCounter
+}
+
+// Float64ObservableCounter is an OpenTelemetry ObservableCounter used to record
+// float64 measurements. It produces no telemetry.
+type Float64ObservableCounter struct {
+	instrument.Float64ObservableCounter
+}
+
+// Int64ObservableGauge is an OpenTelemetry ObservableGauge used to record
+// int64 measurements. It produces no telemetry.
+type Int64ObservableGauge struct {
+	instrument.Int64ObservableGauge
+}
+
+// Float64ObservableGauge is an OpenTelemetry ObservableGauge used to record
+// float64 measurements. It produces no telemetry.
+type Float64ObservableGauge struct {
+	instrument.Float64ObservableGauge
+}
+
+// Int64ObservableUpDownCounter is an OpenTelemetry ObservableUpDownCounter
+// used to record int64 measurements. It produces no telemetry.
+type Int64ObservableUpDownCounter struct {
+	instrument.Int64ObservableUpDownCounter
+}
+
+// Float64ObservableUpDownCounter is an OpenTelemetry ObservableUpDownCounter
+// used to record float64 measurements. It produces no telemetry.
+type Float64ObservableUpDownCounter struct {
+	instrument.Float64ObservableUpDownCounter
+}
+
+// Int64Observer is a recorder of int64 measurements that performs no operation.
+type Int64Observer struct{ instrument.Int64Observer }
+
+// Observe performs no operation.
+func (Int64Observer) Observe(int64, ...attribute.KeyValue) {}
+
+// Float64Observer is a recorder of float64 measurements that performs no
+// operation.
+type Float64Observer struct{ instrument.Float64Observer }
```

Comment:
Looks like we should remove the embedding of `instrument.Float64Observer`

Reply:
Updated. PTAL.